builder/assignmentEnvImageBuilder.go

```diff
@@ -224,14 +251,20 @@ func (assgnEnv *assignmentEnvironment) build() error {
 	}
 }
 
-func (assgnEnv *assignmentEnvironment) publish() error {
+// publish pushes the built image to docker hub if it is not already present
```

Comment:
**// publish the built image ...**

rewrite this comment as shown below.
```go
// publish the built image to docker hub, if required, if it is not already present.
```
Also, why not name the function publishImage(...) similar to pullImage(...)?

Reply:
Done. Renamed publish() to publishImage(). Also got rid of pushImage(), it was not really needed.